storage/src/test/java/tech/pegasys/artemis/storage/ChainStorageClientTest.java

```diff
@@ -137,64 +112,39 @@ public void storageBackedClient_storeInitializeViaNewGenesisState()
     assertThat(client.get().getStore()).isEqualTo(genesisStore);
   }
 
-  @Test
-  public void storageBackedClient_storeInitializeViaStoreInitializedEvent()
-      throws ExecutionException, InterruptedException {
-    final EventBus eventBus = new EventBus();
-    final List<GetStoreRequest> getStoreRequests =
-        EventSink.capture(eventBus, GetStoreRequest.class);
-    final SafeFuture<ChainStorageClient> client =
-        ChainStorageClient.storageBackedClient(eventBus, storageUpdateChannel);
-
-    // We should have posted a request to get the store from storage
-    assertThat(getStoreRequests.size()).isEqualTo(1);
-    // Client shouldn't be initialized yet
-    assertThat(client).isNotDone();
-
-    // Post initialization event
-    final Store genesisStore = Store.get_genesis_store(INITIAL_STATE);
-    eventBus.post(new StoreInitializedFromStorageEvent(Optional.of(genesisStore)));
-    assertThat(client).isCompleted();
-    assertStoreInitialized(client.get());
-    assertStoreIsSet(client.get());
-    assertThat(client.get().getStore()).isEqualTo(genesisStore);
-
-    // Post getStore response - which shouldn't change the store
-    final BeaconState otherState = dataStructureUtil.randomBeaconState(UnsignedLong.ZERO);
-    assertThat(otherState).isNotEqualTo(INITIAL_STATE);
-    final Store otherStore = Store.get_genesis_store(otherState);
-    eventBus.post(new GetStoreResponse(getStoreRequests.get(0).getId(), Optional.of(otherStore)));
-    assertStoreIsSet(client.get());
-    assertThat(client.get().getStore()).isEqualTo(genesisStore);
-  }
-
-  @Test
-  public void storageBackedClient_storeInitializeViaStoreInitializedEvent_emptyStore()
-      throws ExecutionException, InterruptedException {
-    final EventBus eventBus = new EventBus();
-    final List<GetStoreRequest> getStoreRequests =
-        EventSink.capture(eventBus, GetStoreRequest.class);
-    final SafeFuture<ChainStorageClient> client =
-        ChainStorageClient.storageBackedClient(eventBus, storageUpdateChannel);
-
-    // We should have posted a request to get the store from storage
-    assertThat(getStoreRequests.size()).isEqualTo(1);
-    // Client shouldn't be initialized yet
-    assertThat(client).isNotDone();
-
-    // Post initialization event
-    eventBus.post(new StoreInitializedFromStorageEvent(Optional.empty()));
-    assertThat(client).isCompleted();
-    assertStoreNotInitialized(client.get());
-    assertThat(client.get().getStore()).isNull();
-
-    // Now set the genesis state
-    final Store genesisStore = Store.get_genesis_store(INITIAL_STATE);
-    client.get().initializeFromGenesis(INITIAL_STATE);
-    assertStoreInitialized(client.get());
-    assertStoreIsSet(client.get());
-    assertThat(client.get().getStore()).isEqualTo(genesisStore);
-  }
+  // TODO: figure out how to simulate a timeout and uncomment this test
+  //  @Test
+  //  public void storageBackedClient_storeInitializeViaGetStoreRequestAfterTimeout()
```

Comment:
Timeout is simulated by returning a `failedFuture(new TimeoutException())`.  Note that you'll need to use the multiple return pattern of mockito so it doesn't keep returning the timeout future and fail forever so:

`when(storageUpdateChannel.onStoreRequest()).thenReturn(SafeFuture.failedFuture(new TimeoutException()).thenReturn(storeRequestFuture)`

Reply:
Thanks for the tip, was really struggling to find how to simulate a TimeoutException.

Reply:
Done.